hooks/useFetchProjects.ts

```diff
@@ -58,14 +59,45 @@ interface CachedProjects {
 const cacheKey = 'projectsData';
 const apiEndpoint = '/api/projects'; // The new API endpoint
 
+const extractTagsFromProjects = (projects: RepoItem[]) => {
+  return projects.reduce((acc: string[], project) => {
+    const tags = project.languages?.edges?.map(edge => edge.node.name);
+    tags.forEach(tag => {
+      if (!acc.includes(tag)) acc.push(tag);
+    });
+    return acc;
+  }, []);
+};
+
+const filterProjectsByTags = (projects: RepoItem[], tags: ProjectFilter[]) => {
+  if (!projects) return [];
+  if (!tags || tags.length === 0) return projects;
+
+  const projectsFiltered = projects.filter(project => {
+    const projectTags = project.languages?.edges?.map(edge => edge.node.name);
+    return projectTags?.some(tag =>
+      tags.some(t => t.name === tag && t.isActive)
+    );
+  });
+  console.log('🚀 ~ projectsFiltered ~ projectsFiltered:', {
+    projectsFiltered,
+  });
```

Comment:
Do we need this `console.log` in production?

Reply:
you are correct!
log removed.